### DISCUSSION TITLE
How to make custom layout using rowConstraints?

### DISCUSSION THREAD

**Discussion by AndrasE** [igordanchenko/react-photo-album#174] [Q&A] [answered]
Hi Igor, thank you for these amazing packages! ⭐
It's probably a noob question; sorry about it! I'm trying to implement a custom row layout. 
I would like to have the images in a centred container with a max width of 1200px and have max 4 pictures on one row. If the screen is below 600px I would like to have only two pictures per row. I've read the documentation, but I never could figure out how to use the rowConstraints to achieve this. I'm not using TS; I thought I was mentioning it in case it matters.

**Comment by igordanchenko** [comment, marked as answer]
Hi there! Please see if the following settings provide the layout you are looking for.

```jsx
<PhotoAlbum
  layout="rows"
  photos={photos}
  componentsProps={{
    containerProps: { style: { maxWidth: 1200, marginInline: 'auto' } },
  }}
  rowConstraints={(containerWidth) => ({
    maxPhotos: containerWidth > 600 ? 4 : 2,
  })}
/>
```

**Reply by AndrasE** [reply]
Thanks a lot, you're an absulute hero! 🦸
